### PR TITLE
[AMDGPU][GFX12] Add tests for flat_atomic_pk

### PR DESCRIPTION
**Pull request by mariusz-sikora-at-amd**

### CODE REVIEW

File: llvm/test/MC/AMDGPU/gfx12_asm_vimage.s [hunk 1]
```diff
@@ -1018,6 +1018,60 @@ image_atomic_max_flt v[254:255], [v4, v5, v6, v7], s[96:103] dmask:0x3 dim:SQ_RS
 image_atomic_max_flt v0, v0, s[0:7] dmask:0x1 dim:SQ_RSRC_IMG_1D th:TH_ATOMIC_NT
 // GFX12: encoding: [0x00,0x40,0x61,0xd0,0x00,0x00,0x20,0x00,0x00,0x00,0x00,0x00]
 
+image_atomic_pk_add_f16 v0, v0, s[0:7] dmask:0x1 dim:SQ_RSRC_IMG_1D
```
Comment (jayfoad) [inline llvm/test/MC/AMDGPU/gfx12_asm_vimage.s:1021]: Don't need this - these tests are already present starting at line 886.

File: llvm/test/MC/Disassembler/AMDGPU/gfx12_dasm_vimage.txt [hunk 2]
```diff
@@ -1018,6 +1018,60 @@
 # GFX12: image_atomic_max_flt v0, v0, s[0:7] dmask:0x1 dim:SQ_RSRC_IMG_1D th:TH_ATOMIC_NT ; encoding: [0x00,0x40,0x61,0xd0,0x00,0x00,0x20,0x00,0x00,0x00,0x00,0x00]
 0x00,0x40,0x61,0xd0,0x00,0x00,0x20,0x00,0x00,0x00,0x00,0x00
 
+# GFX12: image_atomic_pk_add_f16 v0, v0, s[0:7] dmask:0x1 dim:SQ_RSRC_IMG_1D ; encoding: [0x00,0x80,0x61,0xd0,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00]
```
Comment (jayfoad) [inline llvm/test/MC/Disassembler/AMDGPU/gfx12_dasm_vimage.txt:1021]: Same.

File: llvm/test/MC/AMDGPU/gfx12_asm_vimage.s (hunk 1 above)
Comment (mariusz-sikora-at-amd) [reply llvm/test/MC/AMDGPU/gfx12_asm_vimage.s]: Thanks!, removed from PR

**Review by jayfoad — APPROVED** [review]
LGTM, thanks. Don't forget to remove "image" from the description when you squash-and-merge.

**Pull request merged**